sljit_src/sljitUtils.c

```diff
@@ -82,14 +86,22 @@ static SLJIT_INLINE void allocator_grab_lock(void)
 
 #include <fcntl.h>
 
+#ifdef O_CLOEXEC
+#define SLJIT_CLOEXEC	O_CLOEXEC
+#else /* !O_CLOEXEC */
+#define SLJIT_CLOEXEC	0
+#endif /* O_CLOEXEC */
+
 /* Some old systems do not have MAP_ANON. */
-static sljit_s32 dev_zero = -1;
+static int dev_zero = -1;
 
 #if (defined SLJIT_SINGLE_THREADED && SLJIT_SINGLE_THREADED)
 
-static SLJIT_INLINE sljit_s32 open_dev_zero(void)
+static SLJIT_INLINE int open_dev_zero(void)
 {
-	dev_zero = open("/dev/zero", O_RDWR);
+	if (SLJIT_UNLIKELY(dev_zero < 0))
```

Comment:
this could be taken out, which then makes this change mainly about adding O_CLOEXEC